config.go

```diff
@@ -91,71 +104,86 @@ func (c *Configuration) load(reload ...bool) *Configuration {
 			if err != nil {
 				log.Fatalf("config.load(): %v", err)
 			}
-			c.Resources = append(c.Resources, rc.Resources...)
+			rc.File = resourceConfig.Name()
+			scf.Config = append(scf.Config, rc)
 		}
 	}
 
 	// load resources
-	for _, resource := range c.Resources {
-		switch strings.ToLower(resource.Cloud) {
-		case "azure":
-			switch strings.ToLower(resource.Type) {
-			case "frontdoor":
-				var fd AzureFrontDoor
-				fd.SubscriptionId = resource.SubscriptionId
-				fd.ResourceGroup = resource.ResourceGroup
-				fd.PolicyName = resource.PolicyName
-				fd.IPWhiteList = resource.IPWhiteList
-				fd.Group = resource.Group
-				fd.new(fd)
-			case "storageaccount":
-				var st AzureStorageAccount
-				st.SubscriptionId = resource.SubscriptionId
-				st.ResourceGroup = resource.ResourceGroup
-				st.Name = resource.Name
-				st.IPWhiteList = resource.IPWhiteList
-				st.Group = resource.Group
-				st.new(st)
-			case "keyvault":
-				var kv AzureKeyVault
-				kv.SubscriptionId = resource.SubscriptionId
-				kv.ResourceGroup = resource.ResourceGroup
-				kv.Name = resource.Name
-				kv.IPWhiteList = resource.IPWhiteList
-				kv.Group = resource.Group
-				kv.new(kv)
-			case "postgres":
-				var pg AzurePostgresServer
-				pg.SubscriptionId = resource.SubscriptionId
-				pg.ResourceGroup = resource.ResourceGroup
-				pg.Name = resource.Name
-				pg.IPWhiteList = resource.IPWhiteList
-				pg.Group = resource.Group
-				pg.new(pg)
-			case "redis":
-				var rc AzureRedisCache
-				rc.SubscriptionId = resource.SubscriptionId
-				rc.ResourceGroup = resource.ResourceGroup
-				rc.Name = resource.Name
-				rc.IPWhiteList = resource.IPWhiteList
-				rc.Group = resource.Group
-				rc.new(rc)
-			case "cosmosdb":
-				var cd AzureCosmosDb
-				cd.SubscriptionId = resource.SubscriptionId
-				cd.ResourceGroup = resource.ResourceGroup
-				cd.Name = resource.Name
-				cd.IPWhiteList = resource.IPWhiteList
-				cd.Group = resource.Group
-				cd.new(cd)
+	for _, config := range scf.Config {
+		subId := ""
+		rg := ""
+		for _, resource := range config.Resources {
+			if resource.SubscriptionId == "" {
+				subId = config.Defaults[0].SubscriptionId
```

Comment:
```suggestion
				subId = config.Defaults.SubscriptionId
```
Don't think this needs to be an array